src/main/java/carpetaddonsnotfound/mixins/PlayerEntity_PortalSpectatorDelayMixin.java

```diff
@@ -0,0 +1,23 @@
+package carpetaddonsnotfound.mixins;
+
+import carpetaddonsnotfound.CarpetAddonsNotFoundSettings;
+import net.minecraft.entity.player.PlayerEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+// A priority less than 1000 is required for this mixin to be loaded before base carpet.
```

Comment:
What happens if this is loaded after base carpet? Just curious 🙂

Reply:
This is the mixin in base carpet:

```java
@Inject(method = "getPortalWaitTime()I", at = @At("HEAD"), cancellable = true)
private void onMaxNetherPortalTime(CallbackInfoReturnable<Integer> cir) {
    if(CarpetSettings.portalCreativeDelay != 1 && this.abilities.invulnerable) cir.setReturnValue(CarpetSettings.portalCreativeDelay);
    else if(CarpetSettings.portalSurvivalDelay != 80 && !this.abilities.invulnerable) cir.setReturnValue(CarpetSettings.portalSurvivalDelay);
}
```

So unless our mixin is loaded first, the spectator player will be given the survival timeout.

Reply:
Have you tried to mixin this mixin method?

so something like @Mixin(carpet.creativePortalDelay.class)

Reply:
> Have you tried to mixin this mixin method?
> 
> so something like @mixin(carpet.creativePortalDelay.class)

Is that even safe to do?

Reply:
I don't think it's even possible to inject into another mixin.